### PR TITLE
Updates map_drops.yml 

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: N/A

* **Server Mode**: 

* **Description of Pull Request**: 

Adjusted the rate in map_drops.yml to 1/100000 instead of 1/10000

### CODE REVIEW

File: src/map/mob.cpp [hunk 1]
```diff
@@ -2904,8 +2900,8 @@ int mob_dead(struct mob_data *md, struct block_list *src, int type)
 		if( mapdrops != nullptr ){
 			// Process map wide drops
 			for( const auto& it : mapdrops->globals ){
-				if( rnd_chance( it.second->rate, 10000 ) ){
-					mob_item_drop( md, dlist, mob_setdropitem( it.second.get(), 1, md->mob_id ), 0, it.second->rate, homkillonly || merckillonly );
+				if( rnd_value( 0, 100000 ) < it.second->rate ){
+					mob_item_drop( md, dlist, mob_setdropitem( it.second.get(), 1, md->mob_id ), 0, it.second->rate, (homkillonly || merckillonly), true );
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:2904]: I would prefer multiplying by 10 at these two calls and not touch the other function at all.
The mob_item_drop function is already unreadable enough with all those short ifs... :-(

File: src/map/mob.cpp [hunk 2]
```diff
@@ -2904,8 +2900,8 @@ int mob_dead(struct mob_data *md, struct block_list *src, int type)
 		if( mapdrops != nullptr ){
 			// Process map wide drops
 			for( const auto& it : mapdrops->globals ){
-				if( rnd_chance( it.second->rate, 10000 ) ){
-					mob_item_drop( md, dlist, mob_setdropitem( it.second.get(), 1, md->mob_id ), 0, it.second->rate, homkillonly || merckillonly );
+				if( rnd_value( 0, 100000 ) < it.second->rate ){
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:2903]: Can we add a uint32 variant of the rnd_chance function instead?

File: src/common/random.hpp [hunk 3]
```diff
@@ -11,5 +11,6 @@ void rnd_init(void);
 int32 rnd(void);// [0, SINT32_MAX]
 int32 rnd_value(int32 min, int32 max);// [min, max]
 bool rnd_chance( uint16 chance, uint16 base );
+bool rnd_chance32( int32 chance, int32 base );
```
Comment (Lemongrass3110) [inline src/common/random.hpp:14]:
```suggestion
bool rnd_chance( uint32 chance, uint32 base );
```
C++ has function overloading.

**Review by Lemongrass3110 — APPROVED** [review]
You still need to add parsing logic. It is still capped to uint16rate there!

File: src/common/random.hpp (hunk 3 above)
Comment (Atemo) [reply src/common/random.hpp:14]: `rnd_chance` is using `rnd_value` (`int32 rnd_value(int32 min, int32 max);`) leaving it in int32 was less annoying :(

The compiler shows me a warning from `rnd_chance` in `skill.cpp` when using the overloading, so I renamed the function. Not sure how to fix the warning

File: src/common/random.cpp [hunk 4]
```diff
@@ -46,3 +46,7 @@ int32 rnd_value( int32 min, int32 max ){
 bool rnd_chance( uint16 chance, uint16 base ){
 	return rnd_value( 0, base ) < chance;
 }
+
+bool rnd_chance32( int32 chance, int32 base ){
+	return rnd_value( 0, base ) < chance;
+}
```
Comment (Lemongrass3110) [inline src/common/random.cpp:52]:
```suggestion
```

File: src/common/random.hpp (hunk 3 above)
Comment (Lemongrass3110) [inline src/common/random.hpp:14]:
```suggestion

template <typename T> bool rnd_chance( T chance, T base ){
	return rnd_value( 0, base ) < chance;
}
```

File: src/map/mob.cpp [hunk 5]
```diff
@@ -2904,8 +2904,10 @@ int mob_dead(struct mob_data *md, struct block_list *src, int type)
 		if( mapdrops != nullptr ){
 			// Process map wide drops
 			for( const auto& it : mapdrops->globals ){
-				if( rnd_chance( it.second->rate, 10000 ) ){
-					mob_item_drop( md, dlist, mob_setdropitem( it.second.get(), 1, md->mob_id ), 0, it.second->rate, homkillonly || merckillonly );
+				if( rnd_chance32( it.second->rate, 100000 ) ){
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:2907]:
```suggestion
				if( rnd_chance32( it.second->rate, 100000u ) ){
```

File: src/map/mob.cpp [hunk 6]
```diff
@@ -2914,8 +2916,10 @@ int mob_dead(struct mob_data *md, struct block_list *src, int type)
 
 			if( specific != mapdrops->specific.end() ){
 				for( const auto& it : specific->second ){
-					if( rnd_chance( it.second->rate, 10000 ) ){
-						mob_item_drop( md, dlist, mob_setdropitem( it.second.get(), 1, md->mob_id ), 0, it.second->rate, homkillonly || merckillonly );
+					if( rnd_chance32( it.second->rate, 100000 ) ){
```
Comment (Lemongrass3110) [inline src/map/mob.cpp:2919]:
```suggestion
					if( rnd_chance32( it.second->rate, 100000u ) ){
```

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**